moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -413,6 +417,11 @@ void PlanningSceneMonitor::scenePublishingThread()
       planning_scene_publisher_.publish(msg);
       if (is_full)
         ROS_DEBUG_NAMED(LOGNAME, "Published full planning scene: '%s'", msg.name.c_str());
+      if(publish_planning_scene_frequency_updated_) {
+        ROS_DEBUG_STREAM_NAMED(LOGNAME, "Updating planning scene publish frequency to " << publish_planning_scene_frequency_);
+        rate = ros::Rate(publish_planning_scene_frequency_);
+        publish_planning_scene_frequency_updated_ = false;
+      }
       rate.sleep();
```

Comment:
Please sleep *before* updating the rate to avoid a single longer cycle on change.